examples_tests/35.TAA/main.cpp

```diff
@@ -294,17 +289,40 @@ int main()
     TAAMaterial.setTexture(3u, velocityBuf);
     TAAMaterial.TextureLayer[3].SamplingParams = sparams;
     TAAMaterial.MaterialType = TAAMaterialType;
+    TAAMaterial.ZBuffer = video::ECFN_ALWAYS;
+    TAAMaterial.ZWriteEnable = false;
 
 	uint64_t lastFPSTime = 0;
 
+    core::matrix4SIMD prevVP, currVP;
 	while(!quit && device->run())
 	{
 		driver->beginScene(false, false, video::SColor(255,255,255,255) );
 
         const float clearVel[4]{ 0.f, 0.f, 0.f, 0.f };
-        const float clearColor[4]{ 0.f, 0.f, 1.f, 1.f };
+        const float clearColor[4]{ 0.9f, 0.9f, 0.9f, 1.f };
         const float clearDepth = 0.f;
 
+        //update prevVP and currVP before velocity pass
+        if (g_FrameNum==0u)
+        {
+            currVP = prevVP = driver->getTransform(video::EPTS_PROJ_VIEW);
+        }
+        else
+        {
+            prevVP = currVP;
+            currVP = driver->getTransform(video::EPTS_PROJ_VIEW);
+#if USE_JITTER
+            core::matrix4SIMD jitter;
+            float jitterOffset[3]{ 0.f,0.f,0.f };
+            memcpy(jitterOffset, &JITTER_OFFSETS[g_FrameNum%JITTER_OFFSET_CNT].X, 8);
+            jitter.setTranslation(jitterOffset);
+            currVP = core::matrix4SIMD::concatenateBFollowedByA(jitter, currVP);
+#endif//USE_JITTER
+        }
+        velBufCb->setPrevVP(prevVP);
+        velBufCb->setCurrentVP(currVP);
+
         // VELOCITY BUF PASS
```

Comment:
for the love of god precompute your FBO attachment combos + use Multiple Render Targets to output velocity + color in one pass!